### PR TITLE
deps: use melange from opam

### PR DESCRIPTION
**Pull request by emillon**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,24 +71,14 @@ install-ocamlformat:
...
-dev-deps: melange
+dev-deps:
 	opam install -y $(TEST_DEPS)
 
 .PHONY: coverage-deps
 coverage-deps:
 	opam install -y bisect_ppx
 
-.PHONY: dev-deps-sans-melange
-dev-deps-sans-melange:
```
Comment (jchavarri) [inline Makefile:89]: This command is used here

https://github.com/ocaml/dune/blob/bd1193de7b5b59770d01d0a3f43e6936c6271a16/ci/build-test.sh#L120

Same with `test-all-sans-melange` below.

**Review by jchavarri — APPROVED** [review]
Mentioned in the inline comment: I think there is some cleaning that could be done in `build-test.sh`? Not sure why the tests are not failing, maybe it's not used.

There is also some pinning happening in nix flakes, it would be nice to keep that workflow in sync with the opam one, but I have no clue how to fix those.

File: Makefile (hunk 1 above)
Comment (emillon) [reply Makefile:89]: I updated this by following the comments. But it's not used since we only use that file on windows.

**Comment by emillon** [comment]
@anmonteiro I tried to update the config so that we pick melange >= 1.0.0 from opam instead of pinning. Is there anything to do in flake.nix?

**Comment by Alizter** [comment]
@emillon https://github.com/ocaml/dune/pull/7874 takes care of that I think.

**Review by anmonteiro — APPROVED** [review]
LGTM, the nix changes can be done separately

**Pull request merged**